### PR TITLE
[bindgen] wlr_gles2_renderer

### PR DESCRIPTION
**Pull request by Shinyzenith**
* Note: I have only bound what I need for now.

Signed-off-by: Shinyzenith <aakashsensharma@gmail.com>

### CODE REVIEW

File: src/render/renderer.zig [hunk 1]
```diff
@@ -2,12 +2,15 @@ const wlr = @import("../wlroots.zig");
...
 const wl = wayland.server.wl;
+const Texture = @import("texture.zig");
 
 pub const Renderer = extern struct {
     const Impl = opaque {};
+    const Egl = opaque {};
 
     impl: *const Impl,
 
+    format: *const Impl,
```
Comment (ifreund) [inline src/render/renderer.zig:13]: why is this here?

File: src/render/texture.zig [hunk 2]
```diff
@@ -3,6 +3,16 @@ const wlr = @import("../wlroots.zig");
 const wayland = @import("wayland");
 const wl = wayland.server.wl;
 
+pub const TextureAttribs = extern struct {
+    const GLenum = opaque {};
+    const GLuint = opaque {};
```
Comment (ifreund) [inline src/render/texture.zig:8]: why are these opaque?

File: src/render/texture.zig [hunk 3]
```diff
@@ -3,6 +3,16 @@ const wlr = @import("../wlroots.zig");
...
+
+    target: *const GLenum,
+    tex: *const GLuint,
```
Comment (ifreund) [inline src/render/texture.zig:11]: why are these pointers?

File: src/render/renderer.zig (hunk 1 above)
Comment (Shinyzenith) [reply src/render/renderer.zig]: Apologies, testing code that got pushed by mistake.

File: src/render/texture.zig (hunk 2 above)
Comment (Shinyzenith) [reply src/render/texture.zig:8]: > why are these opaque?

```
typedef unsigned int GLenum;
typedef unsigned int GLuint;
```
 I wasn't sure how I'd wrap this so I just used opaque, should I declare them as u32?

File: src/render/texture.zig (hunk 3 above)
Comment (Shinyzenith) [reply src/render/texture.zig]: > why are these pointers?

Copy paste error, removed.

File: src/render/texture.zig (hunk 2 above)
Comment (ifreund) [reply src/render/texture.zig:8]: if they are 32 bit unsigned integers regardless of target architecture `u32` would be the correct type. I don't know if that's the case and don't really have time to look into it in depth.

Comment (novakne) [reply src/render/texture.zig:8]: It seems that's the case https://www.khronos.org/opengl/wiki/OpenGL_Type

Comment (Shinyzenith) [reply src/render/texture.zig]: Changed the u32.

Comment (ifreund) [reply src/render/texture.zig]: If that is in fact the case I would expect to see `khronos_uint32_t` in the source code not `unsigned int`

Comment (Shinyzenith) [reply src/render/texture.zig]: It's still a unsigned 32bit integer right? which is equivalent to u32 so I assume the patch would work?

Comment (ifreund) [reply src/render/texture.zig]: `unsigned int` is not 32 bits on all targets.

Comment (Shinyzenith) [reply src/render/texture.zig]: > `unsigned int` is not 32 bits on all targets.

Understood, in that case what should I annotate it with?
